### ISSUE TITLE
Issue: Cannot use softmax_type='learnable' with FusedAttention backend

### ISSUE THREAD

**Issue by jootanehorror** [NVIDIA/TransformerEngine#2318]
I'm trying to use `softmax_type='learnable'` with the FusedAttention backend in Transformer Engine, but the system automatically falls back to UnfusedDotProductAttention even when FusedAttention is explicitly enabled.
TE version 2.8.0, torch 2.9.0

```
import os
os.environ["NVTE_FUSED_ATTN"] = "1"
os.environ["NVTE_FLASH_ATTN"] = "0"
os.environ["NVTE_DEBUG"] = "1"
os.environ["NVTE_DEBUG_LEVEL"] = "2"

import torch
from transformer_engine.pytorch.attention import DotProductAttention


attention = DotProductAttention(
    num_attention_heads=8,
    kv_channels=64,
    attention_dropout=0.0,
    softmax_type="learnable",
    qkv_format="bshd"  
).cuda().bfloat16()

q = torch.randn(2, 1024, 8, 64).cuda().bfloat16()
k = torch.randn(2, 1024, 8, 64).cuda().bfloat16() 
v = torch.randn(2, 1024, 8, 64).cuda().bfloat16()

output = attention(q, k, v)

```
```

[DEBUG    | DotProductAttention]: Running with config={'transformer_engine_version': '2.8.0', 'compute_capability': 'sm90', 'flash_attn_version': 'not installed', 'flash_attn_3_version': 'not installed', 'cudnn_version': '9.10.2', 'qkv_type': <class 'torch.Tensor'>, 'qkv_dtype': torch.bfloat16, 'qkv_layout': 'bshd_bshd_bshd', 'batch_size': 2, 'num_heads': 8, 'num_gqa_groups': 8, 'max_seqlen_q': 1024, 'max_seqlen_kv': 1024, 'head_dim_qk': 64, 'head_dim_v': 64, 'attn_mask_type': 'causal', 'window_size': (-1, 0), 'alibi_slopes_shape': None, 'core_attention_bias_type': 'no_bias', 'core_attention_bias_shape': None, 'core_attention_bias_requires_grad': False, 'pad_between_seqs': False, 'attention_dropout': 0.0, 'context_parallel': False, 'cp_comm_type': 'p2p', 'deterministic': False, 'is_training': True, 'fp8': False, 'fp8_meta': {'fp8_checkpoint': False, 'fp8_group': None}, 'inference_params': None, 'softmax_type': 'learnable'}
[DEBUG    | DotProductAttention]: Disabling FlashAttention for softmax_type = learnable
[DEBUG    | DotProductAttention]: Disabling FusedAttention as no backend supports the provided input
[DEBUG    | DotProductAttention]: Available backends = {FlashAttention=False, FusedAttention=False, UnfusedDotProductAttention=True}
[DEBUG    | DotProductAttention]: Selected backend = UnfusedDotProductAttention
[INFO     | DotProductAttention]: Running with UnfusedDotProductAttention backend
```

**Comment by ptrendx** [comment]
Hi @jootanehorror - please update the cuDNN library version. The learnable softmax is only available in cuDNN 9.13 and newer.

**Comment by cyanguwa** [comment]
Please let us know if there are still any issues with learnable softmax. I'll close the issue for now. Please feel free to reopen if there's still any problem. Thanks!
